app/derivative_transformers/kithe/ffmpeg_cli_audio.rb

```diff
@@ -0,0 +1,32 @@
+require 'tty/command'
+module Kithe
+  class FfmpegCliAudio
+    class_attribute :ffmpeg_command, default: "ffmpeg"
+    attr_reader :destination_format
+
+    AUDIO_DERIVATIVE_FORMATS = {
+      standard_webm: { suffix: 'webm', content_type: 'audio/webm',
+        extra_args: ["-ac", "1", "-codec:a", "libopus", "-b:a", "64k"]},
+      standard_mp3:  { suffix: 'mp3',  content_type: 'audio/mpeg',
+        extra_args: ["-ac", "1", "-b:a", "64k"]}
+    }
+
+    def initialize(destination_format)
+      if destination_format.nil?
+        raise ArgumentError.new("No destination format specified.")
+      end
+      @destination_format = destination_format
+    end
+
+    # Will raise TTY::Command::ExitError if the external ffmpeg command returns non-null.
+    def call(original_file)
+      props = AUDIO_DERIVATIVE_FORMATS[destination_format]
+      destination_path = "#{Dir.tmpdir()}/#{destination_format.to_s}.#{props[:suffix]}"
+      ffmpeg_args = [ffmpeg_command, "-i" , original_file.path ] +
+        props[:extra_args] +
+        [destination_path]
+      TTY::Command.new(printer: :null).run(*ffmpeg_args)
+      return File.open(destination_path)
```

Comment:
While this might be fine, look at how the VipsCli one handles the destination file location:

https://github.com/sciencehistory/kithe/blob/master/app/derivative_transformers/kithe/vips_cli_image_to_jpeg.rb#L51

It creates a ruby Tempfile to get a unique location -- which is a bit more straightforward than using that Dir.tmpdir thing. And i've only used Dir.tmpdir as a block, when using it not as a block I'm not sure it's guarantees for cleaning up after itself?

So create a `Tempfile` (not we're passing in a 'base name' and the desired suffix), then pass _that_ path to the command line, then just return the Tempfile. 

Another benefit of the Tempfile appraoch is that ruby promises to delete it from the file system when it goes out of scope. Although kithe _should_ clean up after itself if you return a File too, Tempfile is even more reliable.

Reply:
Also, as a style issue, do not use parens when calling a method with no args. `Dir.tmpdir` not `Dir.tmpdir()` please.

Reply:
Yup - quick change!
Style issue: noted!